types/foundry/common/abstract/backend.d.ts

```diff
@@ -11,94 +17,114 @@ export default abstract class DatabaseBackend {
     /* -------------------------------------------- */
 
     /**
-     * Perform document update operations
-     * @param documentClass    The Document definition
-     * @param context          Context for the requested operation
-     * @param [user]           The requesting User
-     * @returns                The updated Document instances
+     * Update Documents using provided data and context.
+     * It is recommended to use {@link Document.updateDocuments} or {@link Document#update} rather than calling this
+     * method directly.
+     * @param documentClass The Document class definition
+     * @param operation     Parameters of the update operation
+     * @param [user]        The requesting User
+     * @returns An array of updated Document instances
      */
-    update(
-        documentClass: typeof Document,
-        context: DatabaseBackendUpdateContext<Document>,
-        user?: User,
-    ): Promise<Document[]>;
+    update(documentClass: typeof Document, operation: DatabaseUpdateOperation, user?: BaseUser): Promise<Document[]>;
 
-    /** Update primary Document instances */
+    /**
+     * Update Document instances using provided data and operation parameters.
+     * @param documentClass The Document class definition
+     * @param operation     Parameters of the update operation
+     * @param [user]        The requesting User
+     * @returns An array of updated Document instances
+     */
     protected abstract _updateDocuments(
         documentClass: typeof Document,
-        context: DatabaseBackendUpdateContext<Document>,
-        user: BaseUser,
+        operation: DatabaseUpdateOperation,
+        user?: BaseUser,
     ): Promise<Document[]>;
 
     /* -------------------------------------------- */
     /*  Delete Operations                           */
     /* -------------------------------------------- */
 
     /**
-     * Perform document deletion operations
-     * @param documentClass   The Document definition
-     * @param context         Context for the requested operation
-     * @param [user]          The requesting User
-     * @returns               The deleted Document instances
+     * Delete Documents using provided ids and context.
+     * It is recommended to use {@link foundry.abstract.Document.deleteDocuments} or
+     * {@link foundry.abstract.Document#delete} rather than calling this method directly.
+     * @param documentClass The Document class definition
+     * @param operation     Parameters of the delete operation
+     * @param [user]        The requesting User
+     * @returns An array of deleted Document instances
      */
     delete(
-        documentClass: ConstructorOf<Document>,
-        context: DatabaseBackendDeleteContext,
-        user: BaseUser,
-    ): Promise<Document>;
+        documentClass: typeof Document,
+        operation: DatabaseDeleteOperation,
+        user?: BaseUser,
+    ): Promise<Document | undefined>;
```

Comment:
```suggestion
    ): Promise<Document[] | undefined>;
```
A bit weird that it returns undefined if the ids property is undefined.